### DISCUSSION TITLE
Moose example for crystal plasticity modeling under high/low cycle fatigue loading

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#22113] [Q&A Modules: Solid mechanics] [answered]
Dear Moose experts or users,

Are there any developed Moose examples for crystal plasticity modeling under high/low cycle fatigue loading? My purpose is to predict the creep-fatigue life of polycrystal materials and study damage evolution. Any ideas or suggestions about that? Thank you.

**Comment by sapitts** [comment]
Hi @xchengood ,

We do not have any examples of crystal plasticity models being used in fatigue loading within the MOOSE repository currently. I do believe, however, that some research groups may their own branches with these types of models. Perhaps @ngrilli is one of these people?

**Reply by xchengood** [reply]
Hi @sapitts. Thank you for your reply. How do I create a low or high cycle loading in Moose? If I want to build a user object, do you have any advice (e.g. base class)?
@ngrilli, It would be helpful if you could share any built code examples about the creep-fatigue model. Thank you.

**Comment by ngrilli** [comment, marked as answer]
@xchengood an example of backstress implementation is here:

https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/FiniteStrainCrystalPlasticityBackstress.C

which you will need for cyclic fatigue I guess.

An example of creep implementation is here:

https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/CrystalPlasticityDislocationUpdate.C

If you tell us which specific model and material you want to implement, we can help better,
if you are unsure where to start, a Chaboche model may be a good starting point.

Best Regards,
Nicolò
